src/classes/Accounts.cls

```diff
@@ -0,0 +1,219 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Domain class for Accounts in NPSP.
+*/
+public inherited sharing class Accounts extends fflib_SObjects {
+
+    private Map<Id, Account> oldMap = new Map<Id, Account>();
+    private HouseholdNamingService householdNamingService;
+    private HouseholdSettings settings;
+
+    public Accounts(List<Account> records) {
+        this(records, null, null, null);
+    }
+
+    public Accounts(List<Account> records, List<Account> oldlist) {
+        this(records, oldlist, null, null);
+    }
+
+    @TestVisible
+    private Accounts(List<Account> records, List<Account> oldlist, HouseholdNamingService
+            service, HouseholdSettings settings) {
+        super(records, Account.SObjectType);
+        this.oldMap = oldlist == null ? new Map<Id, Account>() : new Map<Id, Account>
+                (oldlist);
+        this.householdNamingService = service == null ? new HouseholdNamingService() : service;
```

Comment:
Yep, this is a good call.  Nice to avoid the unnecessary constructor if we can.  That being said, I just don't love having the null check if it can be avoided.  I just updated these tonight (before reading the comments on the PR) to have the no param constructor pass a new instance to the private constructor, thus eliminating the need to do the null check. 
 Then the test instantiation would be new `HousholdNamingService(new NamingServiceMock())`.  Let me know what you think of that and if you'd still prefer to go this route.  Eventually I'd like to use force-di to inject these collaborator class instances, which will make both of these techniques unnecessary!

Reply:
You're right, its way better to remove the private constructors.  I'll make this change